tests/test_TFNetworkRecLayer.py

```diff
@@ -4161,6 +4161,118 @@ def test_trafo_search_lm():
       assert all(out_seqs[i, :input_seq_lens[i]] == input_seqs[i, :input_seq_lens[i]])
 
 
+def test_offset_no_rec_positional_encoding():
+  # Test `offset` option when `PositionalEncodingLayer` is out of loop.
+  rnd = numpy.random.RandomState(42)
+  n_batch, n_in, n_time, size = 3, 5, 7, 8
+  n_out = n_in
+
+  net_dict = {
+    "input": {"class": "linear", "activation": None, "n_out": size,},
+    "offset": {"class": "constant", "from": [], "value": 1},
+    "pos_enc": {"class": "positional_encoding", "add_to_input": True, "from": ["input"], "offset": "offset"},
+    "output": {"class": "softmax", "target": "data", "from": "pos_enc", "loss": "ce"},
+  }
+
+  config = Config()
+  config.update({
+    "extern_data": {"data": {"dim": n_out, "sparse": True}},
+    "debug_print_layer_output_template": True})
+
+  with make_scope() as session:
+    network = TFNetwork(config=config, train_flag=True)
+    pprint(network.extern_data.data)
+    network.construct_from_dict(net_dict)
+
+    fetches = network.get_fetches_dict()
+    data_input = network.extern_data.data["data"]
+    assert data_input.batch_shape == (None, None)
+
+    train_out = network.get_layer("output").output
+    session.run(tf.variables_initializer(tf.global_variables() + [network.global_train_step]))
+    rand_data = rnd.randint(0, n_out, size=(n_batch, n_time,), dtype="int32")
+    outputs = [train_out.placeholder]
+    info, out = session.run(
+      (fetches, outputs),
+      feed_dict={
+        data_input.placeholder: rand_data,
+        data_input.size_placeholder[0]: numpy.array([n_time] * n_batch, dtype="int32"),
+      })
+    print(info)
+    print(out)  # random...
+
+
+def test_offset_rec_positional_encoding():
```

Comment:
The function should be called `test_PositionalEncodingLayer_offset_in_rec` or so.

Reply:
Done.